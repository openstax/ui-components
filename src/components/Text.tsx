import styled from "styled-components";
import { palette } from "../theme/palette";

export const Heading = styled.h2`
  color: ${palette.neutralDarker}  
  size: 36px;
  font-weight: 700;
`;

export const BodyText = styled.p`
  color: ${palette.neutralDarker};
  size: '18px';
`;
