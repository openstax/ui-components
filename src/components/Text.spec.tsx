import { BodyText, Heading } from './Text';
import renderer from 'react-test-renderer';

describe('Text', () => {
  it('matches body snapshot', () => {
    const tree = renderer.create(
      <BodyText>This is a paragraph</BodyText>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches heading snapshot', () => {
    const tree = renderer.create(
      <Heading>This is a heading</Heading>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
