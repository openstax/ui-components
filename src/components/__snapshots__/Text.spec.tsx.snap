// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text matches body snapshot 1`] = `
<p
  className="sc-eCYdqJ jlZafi"
>
  This is a paragraph
</p>
`;

exports[`Text matches heading snapshot 1`] = `
Array [
  <h2
    className="sc-bczRLJ fZeHof"
  >
    This is an h2
  </h2>,
  <h3
    className="sc-gsnTZi dsraNS"
  >
    This is an h3
  </h3>,
  <h4
    className="sc-dkzDqf ibtgXn"
  >
    This is an h4
  </h4>,
  <h5
    className="sc-hKMtZM jocjt"
  >
    This is an h5
  </h5>,
]
`;
